app/controllers/landing_page_controller.rb

```diff
@@ -110,6 +110,25 @@ def preview
 
   private
 
+  def initialize_i18n!(cid, locale)
+    # This logic is very similar to what we have in ApplicationController
```

Comment:
This could be extracted from the app controller the same way as topbar props and path helpers are extracted to view_utils. Then app controller delegates calls there but the same logic is callable without app controller.